Test app window titles seem to be empty string

### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

I'm trying to write a test to assert that a new window is properly opened. I've initialized a new temp app (`app := test.NewTempApp(t)`), then iterate over `app.Driver().AllWindows()`, but all windows `Title()` method returns an empty string.

To reproduce:

```golang
func Test_testWindow(t *testing.T) {
	a := test.NewTempApp(t)

	_ = a.NewWindow("test")
	_ = a.NewWindow("test")
	_ = a.NewWindow("test")
	_ = a.NewWindow("test")

	for i, w := range a.Driver().AllWindows() {
		t.Log(i, w.Title())
		if w.Title() == "test" {
			return
		}
	}
	t.Error("window not found")
}
```

Outputs:

```
$ go test -v -run=Test_testWindow
=== RUN   Test_testWindow
    about_test.go:24: 0 
    about_test.go:24: 1 
    about_test.go:24: 2 
    about_test.go:24: 3 
    about_test.go:24: 4 
    about_test.go:29: window not found
--- FAIL: Test_testWindow (0.00s)
FAIL
exit status 1
FAIL    gitlab.com/packthathouse/app/internal/app       0.008s
```

### Is it possible to construct a solution with the existing API?

Yes.

### Describe the solution you'd like to see.

I would like to see the window titles returned.

--- a/test/driver.go
+++ b/test/driver.go
@@ -74,7 +74,7 @@ func (d *driver) CanvasForObject(fyne.CanvasObject) fyne.Canvas {
 	return d.windows[len(d.windows)-1].Canvas()
 }
 
-func (d *driver) CreateWindow(string) fyne.Window {
+func (d *driver) CreateWindow(title string) fyne.Window {
 	c := NewCanvas().(*canvas)
 	if d.painter != nil {
 		c.painter = d.painter
@@ -82,7 +82,7 @@ func (d *driver) CreateWindow(string) fyne.Window {
 		c.painter = software.NewPainter()
 	}
 
-	w := &window{canvas: c, driver: d}
+	w := &window{canvas: c, driver: d, title: title}
 
 	d.windowsMutex.Lock()
 	d.windows = append(d.windows, w)

--- a/test/driver_test.go
+++ b/test/driver_test.go
@@ -27,3 +27,10 @@ func Test_driver_AbsolutePositionForObject(t *testing.T) {
 		assert.Equal(t, fyne.NewPos(-2, -3), d.AbsolutePositionForObject(o), "safe area offset (2,3) is subtracted")
 	})
 }
+
+func TestDriver_CreateWindow(t *testing.T) {
+	d := &driver{}
+	w := d.CreateWindow("Test Window")
+
+	assert.Equal(t, "Test Window", w.Title())
+}

